### PR TITLE
Cast masks to np.unit8 before converting to PIL.Image.Image

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

In the recent update to the image segmentation pipeline, the numpy arrays converted to `PIL.Image.Image` with mode `"L"` weren't converted to type `np.uint8`, resulting in corrupted masks. 

Running the following: 
```
import requests
from PIL import Image
from transformers import pipeline

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)
pipe = pipeline("image-segmentation", model="facebook/detr-resnet-50-panoptic")
results = pipe(image)
```

**Output masks on `main`:**
main_0_cat
main_1_couch
main_2_remote
main_3_blanket

**Output masks on this branch**: 
fix_0_cat
fix_1_couch
fix_2_remote
fix_3_blanket

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: tests/pipelines/test_pipelines_image_segmentation.py [hunk 1]
```diff
@@ -229,12 +229,12 @@ def test_small_model_pt_semantic(self):
...
                     "score": None,
                     "label": "LABEL_0",
-                    "mask": "775518a7ed09eea888752176c6ba8f38",
+                    "mask": "42d09072282a32da2ac77375a4c1280f"
                 },
                 {
                     "score": None,
                     "label": "LABEL_1",
-                    "mask": "a12da23a46848128af68c63aa8ba7a02",
+                    "mask": "46b8cc3976732873b219f77a1213c1a5",
```
Comment (amyeroberts) [inline tests/pipelines/test_pipelines_image_segmentation.py:237]: Tests updated to reflect change in masks: 

Input image: 
input_image

Masks on `main`: 

main_label_0
main_label_1




Masks on this branch: 

fix_label_0
fix_label_1

**Review by NielsRogge — APPROVED** [review]
Thanks for fixing!

**Review by sgugger — APPROVED** [review]
Thanks a lot for the quick fix!

**Comment by NielsRogge** [comment]
cc @Narsil, would it be possible to update the inference widgets to include this PR?

We were notified on Twitter that the inference widgets were broken: https://twitter.com/levelsio/status/1580573108431646720?t=d9BlnF9Q2nvRaQFNei6KLw&s=19

**Pull request merged**